### ISSUE TITLE
Investigate and resolve why coreDNS pod fails to start with k8s 1.16.7

### ISSUE THREAD

**Issue by cmoulliard** [snowdrop/k8s-infra#146]
## Issue

Post kubernetes creation deployed using this playbook
```
ansible-playbook ansible/playbook/post_installation.yml \                                                                          
   --limit n01-k116 -b \
   --tags k8s_cluster \
   -e k8s_version=1.16.7 \
   -e coreos_flannel_sha_commit="2140ac876ef134e0ed5af15c65e414cf26827915"
```
we go this error from the coreDNS pod

```
  Type     Reason                  Age                    From               Message
  ----     ------                  ----                   ----               -------
  Warning  FailedScheduling        6m18s (x5 over 6m29s)  default-scheduler  0/1 nodes are available: 1 node(s) had taints that the pod didn't tolerate.
  Normal   Scheduled               6m9s                   default-scheduler  Successfully assigned kube-system/coredns-8c68b4b4d-j5v9k to n01-k116
  Normal   Pulled                  6m8s                   kubelet, n01-k116  Container image "k8s.gcr.io/coredns:1.6.2" already present on machine
  Normal   Created                 6m8s                   kubelet, n01-k116  Created container coredns
  Normal   Started                 6m8s                   kubelet, n01-k116  Started container coredns
  Warning  Unhealthy               5m16s (x6 over 6m6s)   kubelet, n01-k116  Readiness probe failed: HTTP probe failed with statuscode: 503
  Warning  FailedCreatePodSandBox  3m31s                  kubelet, n01-k116  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "33bbfcf3ca436500ca1bc87818b7d56ff25aeeb809ba9a809709456a6e8e72b6" network for pod "coredns-8c68b4b4d-j5v9k": networkPlugin cni failed to set up pod "coredns-8c68b4b4d-j5v9k_kube-system" network: open /run/flannel/subnet.env: no such file or directory
  Warning  FailedCreatePodSandBox  3m29s                  kubelet, n01-k116  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "17ed3214e3eda9bb2b0dce91bc0d9709e74d7d94de4e4b1972d8eac79447b519" network for pod "coredns-8c68b4b4d-j5v9k": networkPlugin cni failed to set up pod "coredns-8c68b4b4d-j5v9k_kube-system" network: open /run/flannel/subnet.env: no such file or directory
  Normal   SandboxChanged          3m28s (x4 over 3m33s)  kubelet, n01-k116  Pod sandbox changed, it will be killed and re-created.
  Warning  FailedCreatePodSandBox  3m28s                  kubelet, n01-k116  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "888b9df29561cb145d48c14ac4a7973949554b9b404984af712995387d0db8e4" network for pod "coredns-8c68b4b4d-j5v9k": networkPlugin cni failed to set up pod "coredns-8c68b4b4d-j5v9k_kube-system" network: open /run/flannel/subnet.env: no such file or directory
  Normal   Pulled                  3m27s                  kubelet, n01-k116  Container image "k8s.gcr.io/coredns:1.6.2" already present on machine
  Normal   Created                 3m27s                  kubelet, n01-k116  Created container coredns
  Normal   Started                 3m27s                  kubelet, n01-k116  Started container coredns
  Warning  Unhealthy               65s (x15 over 3m25s)   kubelet, n01-k116  Readiness probe failed: HTTP probe failed with statuscode: 503
```

**Comment by cmoulliard** [comment]
The problem `networkPlugin cni failed to set up pod "coredns-8c68b4b4d-j5v9k_kube-system" network: open /run/flannel/subnet.env: no such file or directory` is describe here : https://github.com/kubernetes/kubernetes/issues/70202

Maybe a root permission issue ...

**Comment by cmoulliard** [comment]
We can resolve this error by doing within the vm a `sudo kubeadm reset`

**Comment by cmoulliard** [comment]
There is nevertheless still a problem
```
Events:
  Type     Reason            Age                    From               Message
  ----     ------            ----                   ----               -------
  Warning  FailedScheduling  2m37s (x5 over 2m45s)  default-scheduler  0/1 nodes are available: 1 node(s) had taints that the pod didn't tolerate.
  Normal   Scheduled         2m29s                  default-scheduler  Successfully assigned kube-system/coredns-8c68b4b4d-4477n to n01-k116
  Normal   Pulled            2m28s                  kubelet, n01-k116  Container image "k8s.gcr.io/coredns:1.6.2" already present on machine
  Normal   Created           2m28s                  kubelet, n01-k116  Created container coredns
  Normal   Started           2m28s                  kubelet, n01-k116  Started container coredns
  Warning  Unhealthy         2s (x15 over 2m22s)    kubelet, n01-k116  Readiness probe failed: HTTP probe failed with statuscode: 503
```
as the log of the coreDNS reports
```
2020-02-26T09:21:04.926Z [INFO] plugin/reload: Running configuration MD5 = f64cb9b977c7dfca58c4fab108535a76
2020-02-26T09:21:04.926Z [INFO] CoreDNS-1.6.2
2020-02-26T09:21:04.926Z [INFO] linux/amd64, go1.12.8, 795a3eb
CoreDNS-1.6.2
linux/amd64, go1.12.8, 795a3eb
2020-02-26T09:21:05.051Z [INFO] plugin/ready: Still waiting on: "kubernetes"
2020-02-26T09:21:10.929Z [ERROR] plugin/errors: 2 4600448915862524131.838977611985415310. HINFO: read udp 10.244.0.2:34467->213.133.98.98:53: i/o timeout
2020-02-26T09:21:11.928Z [ERROR] plugin/errors: 2 4600448915862524131.838977611985415310. HINFO: read udp 10.244.0.2:45517->213.133.99.99:53: read: no route to host
2020-02-26T09:21:12.928Z [ERROR] plugin/errors: 2 4600448915862524131.838977611985415310. HINFO: read udp 10.244.0.2:42002->213.133.100.100:53: read: no route to host
```

**Comment by cmoulliard** [comment]
Is the firewall blocking internal traffic ? @jacobdotcosta

**Comment by cmoulliard** [comment]
Some guys reported the following trick to fix the issue : https://github.com/kubernetes/kubeadm/issues/193#issuecomment-330060848

```
sudo systemctl stop kubelet
sudo systemctl stop docker
sudo iptables --flush
sudo iptables -tnat --flush
sudo systemctl start kubelet
sudo systemctl start docker
```
I tried what they suggested but we got now 
```
2020-02-26T09:29:59.777Z [ERROR] plugin/errors: 2 1961160434266644054.8108370395063036008. HINFO: read udp 10.244.0.4:47416->213.133.100.100:53: i/o timeout
```

**Comment by jacobdotcosta** [comment]
> Is the firewall blocking internal traffic ? @jacobdotcosta

Firewall shouldn't be blocking internal traffic.

**Comment by cmoulliard** [comment]
There is nevertheless an issue as the `CoreDNS` pod cannot access using the IP address (defined within the K8s Service)
```
kc get svc/kubernetes -o yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2020-02-26T09:20:25Z"
  labels:
    component: apiserver
    provider: kubernetes
  name: kubernetes
  namespace: default
  resourceVersion: "148"
  selfLink: /api/v1/namespaces/default/services/kubernetes
  uid: 30c863b2-4781-46a3-ac95-e2bdbdbf251b
spec:
  clusterIP: 10.96.0.1
  ports:
  - name: https
    port: 443
    protocol: TCP
    targetPort: 6443
  sessionAffinity: None
  type: ClusterIP
```

the KubeApi server
```
E0226 09:55:02.805771       1 reflector.go:126] pkg/mod/k8s.io/client-go@v11.0.0+incompatible/tools/cache/reflector.go:94: Failed to list *v1.Service: Get https://10.96.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
2020-02-26T09:55:12.228Z [INFO] plugin/ready: Still waiting on: "kubernetes"
```

**Comment by cmoulliard** [comment]
The problem is due to the `firewalld` service running on centos

**Comment by jacobdotcosta** [comment]
> The problem is due to the `firewalld` service running on centos

In particular, the port 6443 as opening it solves the problem. Nevertheless, probably it should be open internally but not outside.

Short term fix:

```bash
#  firewall-cmd --permanent --add-port=6443/tcp
```

**Comment by cmoulliard** [comment]
Should we keep it open till the playbook is updated @jacobdotcosta ?

**Comment by cmoulliard** [comment]
Resolved with #150. See last comment
